src/p2p/net_node.inl

```diff
@@ -2289,7 +2304,10 @@ namespace nodetool
     CHECK_AND_ASSERT_MES(address_ok, false,
         "Only IPv4 or IPv6 addresses are supported here");
 
-    const epee::net_utils::network_address na = context.m_remote_address;
+    epee::net_utils::network_address na = context.m_remote_address;
+    if (node_data.my_advertise_network_address.get_type_id() != epee::net_utils::address_type::invalid)  {
+      na = node_data.my_advertise_network_address;
+    }
```

Comment:
If you write a small function like this (pseudocode):

```
get_preferred_address(context, node_data) {
    epee::net_utils::network_address na = context.m_remote_address;
    if (node_data.my_advertise_network_address.get_type_id() != epee::net_utils::address_type::invalid)  {
      na = node_data.my_advertise_network_address;
    }
   return na;
}
```
then in the line `2292`, the `na` can remain const:
`const epee::net_utils::network_address na = get_preferred_address(context, node_data)`

Also lines `2476` - `2481`  would be reduced to:

`const auto preferred_na = get_preferred_address(context, node_data)`